test/integration-test/bpf/reloc.bpf.c

```diff
@@ -46,20 +56,32 @@ __attribute__((noinline)) int pointer_global() {
 
 SEC("uprobe") int pointer(void *ctx) { return pointer_global(); }
 
-__attribute__((noinline)) int struct_flavors_global() {
+__noinline int struct_flavors_global() {
   struct relocated_struct_with_scalars s = {1, 2, 3};
+#ifndef TARGET
   if (bpf_core_field_exists(s.a)) {
     return set_output(__builtin_preserve_access_index(s.a));
+#else
+  if (bpf_core_field_exists(s.d)) {
+    return set_output(__builtin_preserve_access_index(s.d));
+#endif
```

Comment:
Shouldn't this be:
```c
if (bpf_core_field_exists(s.a)) {
   return set_output(__builtin_preserve_access_index(s.a));
} else if (bpf_core_field_exists(s.d)) {
  return set_output(__builtin_preserve_access_index(s.d));
} else {
   return 42
}
```

That would return 1 if a exists, 3 if d exists OR `42` if neither a or d exist.
I don't understand what adding the `ifndef TARGET` adds here.

Reply:
Any expression including `s.a` doesn't compile when TARGET is defined because `a` doesn't exist.